### ISSUE TITLE
Rename docker-maven-plugin to moby-maven-plugin

### ISSUE THREAD

**Issue by rhuss** [fabric8io/docker-maven-plugin#762]
Is this the right step now ?

**Comment by nicolaferraro** [comment]
I thought they renamed the upstream project but continue to release the products as "docker x" (even community bits from the moby repo, as "docker CE"). If so, "moby" will be a name known by a tiny audience. I would not change the name if not "forced" to do so.

**Comment by rhuss** [comment]
Some times ago Docker forbid people to have github projects starting with "docker". I've asked via mail whether 'docker-maven-plugin' as name is correct, but never got an answer. So I still think we are on somewhat slippery ice. Another idea was to rename it to container-maven-plugin (but then we should support something in addition to Docker, too ;-)

Lets wait a bit until the dust settles ....

**Comment by kcrawley** [comment]
there is no shortage of knee-jerk reactions at this point. moby is for engineers who are creating entire container ecosystems. docker and the current toolchain is a part of moby. it would make more sense to spin off an make a `linuxkit-maven-plugin` or `swarmkit-maven-plugin` if the need arised.

tl;dr docker is still docker.

**Comment by rhuss** [comment]
I referring to this [issue from last year](https://www.andreas-jung.com/contents/dont-use-docker-in-github-repo-names-or-as-twitter-handle) ([tweet](https://twitter.com/MacYET/status/775535642793086976)) where Docker considers GitHub repo names starting with "docker" not as fair use of the brand. As Docker's current move happens probably also to strengthen the brand "Docker" as a product, I'm still afraid that we are in muddy waters here. 

I sent the following letter last September to legal@docker.com, but never get any response:

```
Hi,

as maintainer of the community project
https://github.com/fabric8io/docker-maven-plugin I have a question
regarding this naming (also because there is some discussion around this
in twittersphere currently). To be honest, I considered the name
"docker-maven-plugin" to be fair and nominative use, which is also used
by around ~50 other projects on GitHub, too
(https://github.com/search?q=docker-maven-plugin)

However, it seems that the naming is against your naming rules for
GitHub repositories.

Could you please clarify whether the name "docker-maven-plugin" can be
used nevertheless ?

And if not, whether the naming rule also applies to the name of the
Maven plugin itself as stored in Maven central
(http://search.maven.org/#search%7Cga%7C1%7Cdocker-maven-plugin) ?

Thanks !
```

At that time we already thought to rename it to `container-maven-plugin`, but wanted to wait for a response first as this means a considerable amount of effort. fyi, the alternative `maven-docker-plugin` is not possible because of Maven rules (which allows only Maven provided plugins to start with `maven-`)

@kcrawley you are probably right, that `moby-maven-plugin` is not the proper choice, but it's far from a knee-jerk reaction as this is a sword of Damocles over this project.

**Comment by rhuss** [comment]
Just a quick poll, please vote thumbs up / down on the following alternatives:

**Comment by rhuss** [comment]
### container-maven-plugin

**Comment by rhuss** [comment]
### moby-maven-plugin

**Comment by rhuss** [comment]
### docker-maven-plugin

(Leave it as it is and wait ....)

**Comment by rhuss** [comment]
Feel free to add you suggestions, too ...

**Comment by kcrawley** [comment]
**maven-docker-plugin**

**Comment by rhuss** [comment]
> fabric8-docker-plugin

@kcrawley Luckily (or unfortunately ;-) ?) we already have a [fabric8-maven-plugin](https://github.com/fabric8io/fabric8-maven-plugin) which is an extension on d-m-p. So I'm afraid this name would confuse people quite a bit (and is not symmetric)

**Comment by kcrawley** [comment]
@rhuss i'm familiar with it. to be fair, that one should really be named `fabric8-kubeshift-plugin` :hurtrealbad:

**Comment by rhuss** [comment]
@kcrawley thanks for suggestion, like the name 'kubeshift' ;-)

Still would like to keep 'maven' somewhere in the name (as I don't know any Maven plugin which does not), so maybe 'kubeshift-maven-plugin' would be a better name, but then still 'kubeshift' sounds a bit like a two headed chimera which btw might be not so far off ;-) Maybe would it be even better to have seperate 'kubernetes-maven-plugin' and 'openshift-maven-plugin' but that's a different story and would be an even bigger move than renaming.
